packages/components/src/SegmentedControl/SegmentedControlOption.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { PropsWithChildren, useId, useMemo } from "react";
+import { useSegmentedControl } from "./SegmentedControlProvider";
+
+interface SegmentedControlOptionProps extends PropsWithChildren {
```

Comment:
Ability to pass a name for the input would be a nice perk so it could be leveraged inside forms but also saved as value of the form.

Reply:
Considering this as a future TODO as discussed, but will not be done in this PR